pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -668,22 +604,31 @@ func (s *DiscoveryServer) loadAssignmentsForClusterIsolated(proxy *model.Proxy,
 	portMap, f := push.ServicePortByHostname[hostname]
 	push.Mutex.Unlock()
 	if !f {
-		// Shouldn't happen here - but just in case fallback
```

Comment:
The problem I am running into is that this comment is a lie. This does happen all the time, at least with the memory registry and I think with split horizon as well, since ServicePortByHostname is not updated unless the registry is kubernetes